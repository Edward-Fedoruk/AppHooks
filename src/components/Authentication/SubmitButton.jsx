import React from 'react'
import PropTypes from 'prop-types'
import Button from '@material-ui/core/Button'
import { withStyles } from '@material-ui/core/styles'

const styles = () => ({
  logIn: {
    marginTop: "34px",
    background: "#35C1CE",
    "&:hover": {
      opacity: ".9",
      background: "#192B81",
    }
  },
})

const SubmitButton = ({ classes, text, styles }) => (
  <Button 
    fullWidth 
    variant="text" 
    color="primary" 
    type="submit"
    className={classes.logIn}
    style={styles}
  >
    { text }
  </Button>
)


SubmitButton.propTypes = {
  text: PropTypes.string.isRequired,
  classes: PropTypes.object.isRequired,
  styles: PropTypes.object
}

export default withStyles(styles)(SubmitButton)
