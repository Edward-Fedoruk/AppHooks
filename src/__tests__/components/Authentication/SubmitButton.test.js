import React from 'react'
import SubmitButton from '../../../components/Authentication/SubmitButton'
import { shallow } from "enzyme"

describe('SubmitButton', () => {
  it('should be defined', () => {
    expect(SubmitButton).toBeDefined()
  })
  it('should render correctly', () => {
    const tree = shallow(
      <SubmitButton text={"test"} />
    )
    expect(tree).toMatchSnapshot()
  })
  it('should have text props', () => {
    const tree = shallow(
      <SubmitButton text={"test"} />
    )
    expect(tree.props().text).toEqual("test")
  })
})