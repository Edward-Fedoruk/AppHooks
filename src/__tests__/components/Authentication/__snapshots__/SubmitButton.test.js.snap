// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton should render correctly 1`] = `
<SubmitButton
  className={Object {}}
  classes={
    Object {
      "logIn": "SubmitButton-logIn-1",
    }
  }
  styles={Object {}}
  text="test"
/>
`;
